File: Data Structures & Algorithms/search-2d-matrix/submission-1.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        n, m = len(matrix), len(matrix[0])
        l, r = 0, n * m - 1

        while l <= r:
            mid = l + (r - l) // 2

            row, col = mid // m, mid % m

            if matrix[row][col] < target:
                l = mid + 1
            elif matrix[row][col] > target:
                r = mid - 1
            else:
                return True


        return False

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int n = numbers.size();
        int lo = 0, hi = n-1;

        while(lo<hi){
            int sum = numbers[lo] + numbers[hi];
            if(sum > target)
                hi--;
            else if(sum < target)
                lo++;
            else
                return {lo+1, hi+1};
        }

        return {};
    }
};
